src/test/ui/proc-macro/attributes-on-modules-fail.rs

```diff
@@ -0,0 +1,27 @@
+// aux-build:test-macros.rs
+
+#[macro_use]
+extern crate test_macros;
+
+#[identity_attr] //~ ERROR custom attributes cannot be applied to modules
+mod m {
+    pub struct X;
+
+    type A = Y; //~ ERROR cannot find type `Y` in this scope
+}
+
+struct Y;
+type A = X; //~ ERROR cannot find type `X` in this scope
+
+#[derive(Copy)] //~ ERROR `derive` may only be applied to structs, enums and unions
+mod n {}
+
+#[empty_attr]
+mod module; //~ ERROR non-inline modules in proc macro input are unstable
+
+#[empty_attr] //~ ERROR custom attributes cannot be applied to modules
+mod outer {
+    mod inner; //~ ERROR non-inline modules in proc macro input are unstable
```

Comment:
A case with an inline module in an inline module with an attribute would be good.